### ISSUE TITLE
Cutoff analysis hitch

### ISSUE THREAD

**Issue by vreuter** [taoliu/MACS#214]
When running `callpeak` with `--cutoff-analysis`, I'm hitting a `KeyError`, with what looks like maybe a negative log of a small probability value. I'm wondering if any others have encountered something similar, and whether there's a known cause/workaround.
```
INFO  @ Fri, 06 Oct 2017 18:32:35: #3 Call peaks... 
INFO  @ Fri, 06 Oct 2017 18:32:35: #3 Pre-compute pvalue-qvalue table... 
INFO  @ Fri, 06 Oct 2017 18:32:35: #3 Cutoff will be automatically decided! 
Traceback (most recent call last):
  File "~/venvs/macs2/bin/macs2", line 617, in <module>
    main()
  File "~/venvs/macs2/bin/macs2", line 57, in main
    run( args )
  File "~/venvs/macs2/local/lib/python2.7/site-packages/MACS2/callpeak_cmd.py", line 264, in run
    peakdetect.call_peaks()
  File "MACS2/PeakDetect.pyx", line 105, in MACS2.PeakDetect.PeakDetect.call_peaks (MACS2/PeakDetect.c:1632)
  File "MACS2/PeakDetect.pyx", line 263, in MACS2.PeakDetect.PeakDetect.__call_peaks_w_control (MACS2/PeakDetect.c:3460)
  File "MACS2/IO/CallPeakUnit.pyx", line 861, in MACS2.IO.CallPeakUnit.CallerFromAlignments.call_peaks (MACS2/IO/CallPeakUnit.c:12334)
  File "MACS2/IO/CallPeakUnit.pyx", line 884, in MACS2.IO.CallPeakUnit.CallerFromAlignments.call_peaks (MACS2/IO/CallPeakUnit.c:11697)
  File "MACS2/IO/CallPeakUnit.pyx", line 850, in MACS2.IO.CallPeakUnit.CallerFromAlignments.__pre_computes (MACS2/IO/CallPeakUnit.c:11115)
KeyError: 9.899999618530273
```

**Comment by davidecarnevali** [comment]
Similar problem here...
I'm running macs2 ver. 2.1.1.20160309
When running the command `macs2 callpeak -t treatment.bam -c input.bam -g hs -q 0.05 --extsize 150 --nomodel --cutoff-analysis -n output` i get the following error:

```
Traceback (most recent call last):
  File "/hpc/home/davide/.local/bin/macs2", line 617, in <module>
    main()
  File "/hpc/home/davide/.local/bin/macs2", line 57, in main
    run( args )
  File "/hpc/home/davide/.local/lib/python2.7/site-packages/MACS2/callpeak_cmd.py", line 264, in run
    peakdetect.call_peaks()
  File "MACS2/PeakDetect.pyx", line 105, in MACS2.PeakDetect.PeakDetect.call_peaks (MACS2/PeakDetect.c:1632)
  File "MACS2/PeakDetect.pyx", line 263, in MACS2.PeakDetect.PeakDetect.__call_peaks_w_control (MACS2/PeakDetect.c:3460)
  File "MACS2/IO/CallPeakUnit.pyx", line 861, in MACS2.IO.CallPeakUnit.CallerFromAlignments.call_peaks (MACS2/IO/CallPeakUnit.c:12334)
  File "MACS2/IO/CallPeakUnit.pyx", line 884, in MACS2.IO.CallPeakUnit.CallerFromAlignments.call_peaks (MACS2/IO/CallPeakUnit.c:11697)
  File "MACS2/IO/CallPeakUnit.pyx", line 850, in MACS2.IO.CallPeakUnit.CallerFromAlignments.__pre_computes (MACS2/IO/CallPeakUnit.c:11145)
ZeroDivisionError: integer division or modulo by zero
```
Running without `--cutoff-analysis` raises no error
Please, help!

**Comment by MikeWLloyd** [comment]
Seeing this same error: 

```
INFO  @ Mon, 20 Apr 2020 09:12:11: #1 read fragment files...
INFO  @ Mon, 20 Apr 2020 09:12:11: #1 read treatment fragments...
INFO  @ Mon, 20 Apr 2020 09:12:11: 50 fragments have been read.
INFO  @ Mon, 20 Apr 2020 09:12:11: #1 mean fragment size is determined as 143.0 bp from treatment
INFO  @ Mon, 20 Apr 2020 09:12:11: #1 fragment size = 143.0
INFO  @ Mon, 20 Apr 2020 09:12:11: #1  total fragments in treatment: 50
INFO  @ Mon, 20 Apr 2020 09:12:11: #1 finished!
INFO  @ Mon, 20 Apr 2020 09:12:11: #2 Build Peak Model...
INFO  @ Mon, 20 Apr 2020 09:12:11: #2 Skipped...
INFO  @ Mon, 20 Apr 2020 09:12:11: #3 Call peaks...
INFO  @ Mon, 20 Apr 2020 09:12:11: #3 Pre-compute pvalue-qvalue table...
INFO  @ Mon, 20 Apr 2020 09:12:11: #3 Cutoff vs peaks called will be analyzed!
KeyError: 9.899999618530273
Exception ignored in: 'MACS2.IO.CallPeakUnit.CallerFromAlignments.__pre_computes'
KeyError: 9.899999618530273
INFO  @ Mon, 20 Apr 2020 09:12:11: #3 Call peaks for each chromosome...
INFO  @ Mon, 20 Apr 2020 09:12:11: #4 Write output xls file... ./testcall_peaks.xls
INFO  @ Mon, 20 Apr 2020 09:12:11: #4 Write peak in narrowPeak format file... ./testcall_peaks.narrowPeak
INFO  @ Mon, 20 Apr 2020 09:12:11: #4 Write summits bed file... ./testcall_summits.bed
INFO  @ Mon, 20 Apr 2020 09:12:11: Done!
```

Admittedly I am testing with a small dummy file, could be related to lack of peaks on certain chromosomes. That we have the same key error with the exact same number is curious to me. Is this in some way hard coded in the tool?

**Comment by taoliu** [comment]
@MikeWLloyd It's not related to chromosome names, but related to p-value -> q-value checkup. I used a python dictionary and it seems to be something like a precision error. The number 9.899999618530273 looks like a 64bit float value and the dictionary may use a 32bit float for key. Could you share me the dummy file for testing? Thanks a lot!

**Comment by MikeWLloyd** [comment]
Very small example I have been using (removed other chr from header): 

```
@HD	VN:1.6	SO:coordinate
@SQ	SN:1	LN:195181283
A00739:196:HNK5JDMXX:2:1438:18367:29543	163	1	3000166	36	82M	=	3000162	77	*	*
A00739:196:HNK5JDMXX:2:1438:18367:29543	83	1	3000162	36	81M	=	3000166	-77	*	*
A00739:196:HNK5JDMXX:2:1162:21296:2738	99	1	3000210	39	97M	=	3000330	216	*	*
A00739:196:HNK5JDMXX:2:1162:21296:2738	147	1	3000330	39	96M	=	3000210	-216	*	*
A00739:196:HNK5JDMXX:2:1153:5620:33426	99	1	3000745	32	97M	=	3000928	279	*	*
A00739:196:HNK5JDMXX:1:1437:30716:17174	163	1	3000903	36	96M	=	3000928	121	*	*
A00739:196:HNK5JDMXX:1:1437:30716:17174	83	1	3000928	36	96M	=	3000903	-121	*	*
A00739:196:HNK5JDMXX:2:1153:5620:33426	147	1	3000928	32	96M	=	3000745	-279	*	*
A00739:196:HNK5JDMXX:2:1145:23439:23390	163	1	3000947	37	97M	=	3000991	140	*	*
A00739:196:HNK5JDMXX:2:1145:23439:23390	83	1	3000991	37	96M	=	3000947	-140	*	*
A00739:196:HNK5JDMXX:1:1487:15700:9831	163	1	3001173	42	97M	=	3001273	196	*	*
A00739:196:HNK5JDMXX:1:1487:15700:9831	83	1	3001273	42	96M	=	3001173	-196	*	*
A00739:196:HNK5JDMXX:2:2488:28818:5682	99	1	3001346	30	71M	=	3001342	66	*	*
A00739:196:HNK5JDMXX:2:2488:28818:5682	147	1	3001342	30	70M	=	3001346	-66	*	*
A00739:196:HNK5JDMXX:1:1432:10050:28134	163	1	3001570	39	92M	=	3001566	87	*	*
A00739:196:HNK5JDMXX:1:1432:10050:28134	83	1	3001566	39	91M	=	3001570	-87	*	*
A00739:196:HNK5JDMXX:1:1358:16098:28369	163	1	3001783	34	97M	=	3001910	223	*	*
A00739:196:HNK5JDMXX:1:1285:4056:32503	99	1	3001817	38	97M	=	3001897	176	*	*
A00739:196:HNK5JDMXX:1:1285:4056:32503	147	1	3001897	38	96M	=	3001817	-176	*	*
A00739:196:HNK5JDMXX:1:1358:16098:28369	83	1	3001910	34	96M	=	3001783	-223	*	*
A00739:196:HNK5JDMXX:1:2486:29658:26021	99	1	3003166	42	96M	=	3003643	544	*	*
A00739:196:HNK5JDMXX:2:2141:26268:28479	163	1	3003225	37	85M	=	3003221	80	*	*
A00739:196:HNK5JDMXX:2:2141:26268:28479	83	1	3003221	37	84M	=	3003225	-80	*	*
A00739:196:HNK5JDMXX:2:2466:30481:18396	163	1	3003242	42	97M	=	3003352	206	*	*
A00739:196:HNK5JDMXX:2:1216:2320:20478	99	1	3003276	42	73M	=	3003272	68	*	*
A00739:196:HNK5JDMXX:2:1216:2320:20478	147	1	3003272	42	72M	=	3003276	-68	*	*
```


The example is being used to debug a genome coordinate conversion tool, which is why it is so small (needed to simply get Macs2 running). 

I had been dealing with a separate Macs2 error related to the output from the conversion tool:

```
Traceback (most recent call last):
  File "/opt/conda/bin/macs2", line 652, in <module>
    main()
  File "/opt/conda/bin/macs2", line 51, in main
    run( args )
  File "/opt/conda/lib/python3.7/site-packages/MACS2/callpeak_cmd.py", line 64, in run
    if options.PE_MODE: (treat, control) = load_frag_files_options (options)
  File "/opt/conda/lib/python3.7/site-packages/MACS2/callpeak_cmd.py", line 350, in load_frag_files_options
    treat = tp.build_petrack()
  File "MACS2/IO/Parser.pyx", line 1120, in MACS2.IO.Parser.BAMPEParser.build_petrack
  File "MACS2/IO/Parser.pyx", line 1153, in MACS2.IO.Parser.BAMPEParser.build_petrack
ZeroDivisionError: float division
```

However, I have determined this error is due to the SAM flags dropping `read mapped in proper pair`. 

So, for example:

```
A00739:196:HNK5JDMXX:2:1438:18367:29543	163	1	3000166	36	82M	=	3000162	77	*	*
A00739:196:HNK5JDMXX:2:1438:18367:29543	83	1	3000162	36	81M	=	3000166	-77	*	*
```

After conversion becomes: 

```
A00739:196:HNK5JDMXX:1:1185:29297:14121 161     1       3003426 42      97M     =       3003679 349     *       *
A00739:196:HNK5JDMXX:1:1185:29297:14121 81      1       3003679 42      96M     =       3003426 -349    *       *
```

I only mention this, because others may encounter a similar issue and I couldn't find any documentation of this issue / resolution.

**Comment by taoliu** [comment]
@MikeWLloyd I converted the SAM file containing 13 pairs you pasted into a BAM file then 

```
$macs2 callpeak -f BAMPE -t dummy.bam --cutoff-analysis
INFO  @ Mon, 20 Apr 2020 10:32:48:
# Command line: callpeak -f BAMPE -t dummy.bam --cutoff-analysis
# ARGUMENTS LIST:
# name = NA
# format = BAMPE
# ChIP-seq file = ['dummy.bam']
# control file = None
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff = 5.00e-02
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 10000 bps
# Broad region calling is off
# Paired-End mode is on

INFO  @ Mon, 20 Apr 2020 10:32:48: #1 read fragment files...
INFO  @ Mon, 20 Apr 2020 10:32:48: #1 read treatment fragments...
INFO  @ Mon, 20 Apr 2020 10:32:48: 13 fragments have been read.
INFO  @ Mon, 20 Apr 2020 10:32:48: #1 mean fragment size is determined as 174.8 bp from treatment
INFO  @ Mon, 20 Apr 2020 10:32:48: #1 fragment size = 174.8
INFO  @ Mon, 20 Apr 2020 10:32:48: #1  total fragments in treatment: 13
INFO  @ Mon, 20 Apr 2020 10:32:48: #1 user defined the maximum fragments...
INFO  @ Mon, 20 Apr 2020 10:32:48: #1 filter out redundant fragments by allowing at most 1 identical fragment(s)
INFO  @ Mon, 20 Apr 2020 10:32:48: #1  fragments after filtering in treatment: 13
INFO  @ Mon, 20 Apr 2020 10:32:48: #1  Redundant rate of treatment: 0.00
INFO  @ Mon, 20 Apr 2020 10:32:48: #1 finished!
INFO  @ Mon, 20 Apr 2020 10:32:48: #2 Build Peak Model...
INFO  @ Mon, 20 Apr 2020 10:32:48: #2 Skipped...
INFO  @ Mon, 20 Apr 2020 10:32:48: #3 Call peaks...
INFO  @ Mon, 20 Apr 2020 10:32:48: #3 Pre-compute pvalue-qvalue table...
INFO  @ Mon, 20 Apr 2020 10:32:48: #3 Cutoff vs peaks called will be analyzed!
ZeroDivisionError: division by zero
Exception ignored in: 'MACS2.IO.CallPeakUnit.CallerFromAlignments.__pre_computes'
Traceback (most recent call last):
  File "/Users/taoliu/Codes/TLPython3/lib/python3.8/site-packages/MACS2-2.2.7.1-py3.8-macosx-10.9-x86_64.egg/MACS2/callpeak_cmd.py", line 256, in run
    peakdetect.call_peaks()
ZeroDivisionError: division by zero
INFO  @ Mon, 20 Apr 2020 10:32:48: #3 Call peaks for each chromosome...
INFO  @ Mon, 20 Apr 2020 10:32:48: #4 Write output xls file... NA_peaks.xls
INFO  @ Mon, 20 Apr 2020 10:32:48: #4 Write peak in narrowPeak format file... NA_peaks.narrowPeak
INFO  @ Mon, 20 Apr 2020 10:32:48: #4 Write summits bed file... NA_summits.bed
INFO  @ Mon, 20 Apr 2020 10:32:48: Done!
```

The zeroDivisionError is due to the fact that there is no peaks can be called under certain cutoff. I will silence the error and skip the corresponding cutoff during cutoff-analysis in the future version. But I can't reproduce the `KeyError` you saw. Could you share me the file you used before which contains 50 pairs?

**Comment by MikeWLloyd** [comment]
Strange. Here was my call on the same file with 13 pairs. Perhaps I am doing something wrong? 

```
singularity run macs2_2.2.6.sif callpeak \
> -f BAMPE \
> --nomodel \
> -g mm \
> --keep-dup all \
> --cutoff-analysis \
> -n testcall \
> -t mini.bam \
> --outdir ./

INFO  @ Mon, 20 Apr 2020 10:40:25:
# Command line: callpeak -f BAMPE --nomodel -g mm --keep-dup all --cutoff-analysis -n testcall -t mini.bam --outdir ./
# ARGUMENTS LIST:
# name = testcall
# format = BAMPE
# ChIP-seq file = ['mini.bam']
# control file = None
# effective genome size = 1.87e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff = 5.00e-02
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 10000 bps
# Broad region calling is off
# Paired-End mode is on

INFO  @ Mon, 20 Apr 2020 10:40:25: #1 read fragment files...
INFO  @ Mon, 20 Apr 2020 10:40:25: #1 read treatment fragments...
INFO  @ Mon, 20 Apr 2020 10:40:25: 13 fragments have been read.
INFO  @ Mon, 20 Apr 2020 10:40:25: #1 mean fragment size is determined as 174.8 bp from treatment
INFO  @ Mon, 20 Apr 2020 10:40:25: #1 fragment size = 174.8
INFO  @ Mon, 20 Apr 2020 10:40:25: #1  total fragments in treatment: 13
INFO  @ Mon, 20 Apr 2020 10:40:25: #1 finished!
INFO  @ Mon, 20 Apr 2020 10:40:25: #2 Build Peak Model...
INFO  @ Mon, 20 Apr 2020 10:40:25: #2 Skipped...
INFO  @ Mon, 20 Apr 2020 10:40:25: #3 Call peaks...
INFO  @ Mon, 20 Apr 2020 10:40:25: #3 Pre-compute pvalue-qvalue table...
INFO  @ Mon, 20 Apr 2020 10:40:25: #3 Cutoff vs peaks called will be analyzed!
KeyError: 9.899999618530273
Exception ignored in: 'MACS2.IO.CallPeakUnit.CallerFromAlignments.__pre_computes'
KeyError: 9.899999618530273
INFO  @ Mon, 20 Apr 2020 10:40:25: #3 Call peaks for each chromosome...
INFO  @ Mon, 20 Apr 2020 10:40:25: #4 Write output xls file... ./testcall_peaks.xls
INFO  @ Mon, 20 Apr 2020 10:40:25: #4 Write peak in narrowPeak format file... ./testcall_peaks.narrowPeak
INFO  @ Mon, 20 Apr 2020 10:40:25: #4 Write summits bed file... ./testcall_summits.bed
INFO  @ Mon, 20 Apr 2020 10:40:25: Done!
```

Here is the file with 50 pairs (sharing as SAM to avoid having to upload a file). The file with 13 pairs is a subset of the 50: 

```
@HD	VN:1.6	SO:coordinate
@SQ	SN:1	LN:195181283
A00739:196:HNK5JDMXX:2:1438:18367:29543	163	1	3000166	36	82M	=	3000162	77	*	*
A00739:196:HNK5JDMXX:2:1438:18367:29543	83	1	3000162	36	81M	=	3000166	-77	*	*
A00739:196:HNK5JDMXX:2:1162:21296:2738	99	1	3000210	39	97M	=	3000330	216	*	*
A00739:196:HNK5JDMXX:2:1162:21296:2738	147	1	3000330	39	96M	=	3000210	-216	*	*
A00739:196:HNK5JDMXX:2:1153:5620:33426	99	1	3000745	32	97M	=	3000928	279	*	*
A00739:196:HNK5JDMXX:1:1437:30716:17174	163	1	3000903	36	96M	=	3000928	121	*	*
A00739:196:HNK5JDMXX:1:1437:30716:17174	83	1	3000928	36	96M	=	3000903	-121	*	*
A00739:196:HNK5JDMXX:2:1153:5620:33426	147	1	3000928	32	96M	=	3000745	-279	*	*
A00739:196:HNK5JDMXX:2:1145:23439:23390	163	1	3000947	37	97M	=	3000991	140	*	*
A00739:196:HNK5JDMXX:2:1145:23439:23390	83	1	3000991	37	96M	=	3000947	-140	*	*
A00739:196:HNK5JDMXX:1:1487:15700:9831	163	1	3001173	42	97M	=	3001273	196	*	*
A00739:196:HNK5JDMXX:1:1487:15700:9831	83	1	3001273	42	96M	=	3001173	-196	*	*
A00739:196:HNK5JDMXX:2:2488:28818:5682	99	1	3001346	30	71M	=	3001342	66	*	*
A00739:196:HNK5JDMXX:2:2488:28818:5682	147	1	3001342	30	70M	=	3001346	-66	*	*
A00739:196:HNK5JDMXX:1:1432:10050:28134	163	1	3001570	39	92M	=	3001566	87	*	*
A00739:196:HNK5JDMXX:1:1432:10050:28134	83	1	3001566	39	91M	=	3001570	-87	*	*
A00739:196:HNK5JDMXX:1:1358:16098:28369	163	1	3001783	34	97M	=	3001910	223	*	*
A00739:196:HNK5JDMXX:1:1285:4056:32503	99	1	3001817	38	97M	=	3001897	176	*	*
A00739:196:HNK5JDMXX:1:1285:4056:32503	147	1	3001897	38	96M	=	3001817	-176	*	*
A00739:196:HNK5JDMXX:1:1358:16098:28369	83	1	3001910	34	96M	=	3001783	-223	*	*
A00739:196:HNK5JDMXX:1:2486:29658:26021	99	1	3003166	42	96M	=	3003643	544	*	*
A00739:196:HNK5JDMXX:2:2141:26268:28479	163	1	3003225	37	85M	=	3003221	80	*	*
A00739:196:HNK5JDMXX:2:2141:26268:28479	83	1	3003221	37	84M	=	3003225	-80	*	*
A00739:196:HNK5JDMXX:2:2466:30481:18396	163	1	3003242	42	97M	=	3003352	206	*	*
A00739:196:HNK5JDMXX:2:1216:2320:20478	99	1	3003276	42	73M	=	3003272	68	*	*
A00739:196:HNK5JDMXX:2:1216:2320:20478	147	1	3003272	42	72M	=	3003276	-68	*	*
A00739:196:HNK5JDMXX:2:2375:23393:20588	163	1	3003293	42	97M	=	3003474	277	*	*
A00739:196:HNK5JDMXX:2:2168:31006:17519	99	1	3003336	42	97M	=	3003471	231	*	*
A00739:196:HNK5JDMXX:2:2466:30481:18396	83	1	3003352	42	96M	=	3003242	-206	*	*
A00739:196:HNK5JDMXX:1:2418:31792:30874	99	1	3003377	42	31M	=	3003373	26	*	*
A00739:196:HNK5JDMXX:1:2418:31792:30874	147	1	3003373	42	30M	=	3003377	-26	*	*
A00739:196:HNK5JDMXX:1:1185:29297:14121	163	1	3003422	42	97M	=	3003679	353	*	*
A00739:196:HNK5JDMXX:2:1343:20320:19179	99	1	3003445	38	97M	=	3003469	120	*	*
A00739:196:HNK5JDMXX:2:1343:20320:19179	147	1	3003469	38	96M	=	3003445	-120	*	*
A00739:196:HNK5JDMXX:2:2168:31006:17519	147	1	3003471	42	96M	=	3003336	-231	*	*
A00739:196:HNK5JDMXX:2:2375:23393:20588	83	1	3003474	42	96M	=	3003293	-277	*	*
A00739:196:HNK5JDMXX:1:2365:9118:13275	163	1	3003482	34	50M	=	3003478	45	*	*
A00739:196:HNK5JDMXX:1:2365:9118:13275	83	1	3003478	34	49M	=	3003482	-45	*	*
A00739:196:HNK5JDMXX:2:1114:12237:33426	163	1	3003506	32	36M	=	3003502	31	*	*
A00739:196:HNK5JDMXX:2:1114:12237:33426	83	1	3003502	32	35M	=	3003506	-31	*	*
A00739:196:HNK5JDMXX:1:2486:29658:26021	147	1	3003643	42	67M	=	3003166	-544	*	*
A00739:196:HNK5JDMXX:2:2214:31132:21778	163	1	3003656	42	94M	=	3003721	161	*	*
A00739:196:HNK5JDMXX:2:2381:15167:18270	163	1	3003681	42	97M	=	3003696	111	*	*
A00739:196:HNK5JDMXX:1:1185:29297:14121	83	1	3003679	42	96M	=	3003422	-353	*	*
A00739:196:HNK5JDMXX:2:2381:15167:18270	83	1	3003696	42	96M	=	3003681	-111	*	*
A00739:196:HNK5JDMXX:2:2214:31132:21778	83	1	3003721	42	96M	=	3003656	-161	*	*
A00739:196:HNK5JDMXX:2:1125:19334:4820	99	1	3003727	42	80M	=	3003723	75	*	*
A00739:196:HNK5JDMXX:2:1125:19334:4820	147	1	3003723	42	79M	=	3003727	-75	*	*
A00739:196:HNK5JDMXX:2:1238:12491:5995	163	1	3003766	42	97M	=	3003837	167	*	*
A00739:196:HNK5JDMXX:2:1238:12491:5995	83	1	3003837	42	96M	=	3003766	-167	*	*
A00739:196:HNK5JDMXX:1:1224:9073:9846	99	1	3003866	42	32M	=	3003862	27	*	*
A00739:196:HNK5JDMXX:1:1224:9073:9846	147	1	3003862	42	31M	=	3003866	-27	*	*
A00739:196:HNK5JDMXX:1:2380:20989:10661	163	1	3003933	42	79M	=	3003929	74	*	*
A00739:196:HNK5JDMXX:1:2380:20989:10661	83	1	3003929	42	78M	=	3003933	-74	*	*
A00739:196:HNK5JDMXX:2:1450:13503:28291	163	1	3004121	42	69M	=	3004117	64	*	*
A00739:196:HNK5JDMXX:2:1450:13503:28291	83	1	3004117	42	68M	=	3004121	-64	*	*
A00739:196:HNK5JDMXX:2:2251:6234:32769	99	1	3004621	42	97M	=	3004662	137	*	*
A00739:196:HNK5JDMXX:2:2204:5475:18490	99	1	3004631	42	97M	=	3004665	130	*	*
A00739:196:HNK5JDMXX:2:2251:6234:32769	147	1	3004662	42	96M	=	3004621	-137	*	*
A00739:196:HNK5JDMXX:2:2204:5475:18490	147	1	3004665	42	96M	=	3004631	-130	*	*
A00739:196:HNK5JDMXX:2:2111:15628:36448	163	1	3004879	42	35M	=	3004875	30	*	*
A00739:196:HNK5JDMXX:2:2111:15628:36448	83	1	3004875	42	34M	=	3004879	-30	*	*
A00739:196:HNK5JDMXX:1:2251:27814:30154	163	1	3004900	42	75M	=	3004896	70	*	*
A00739:196:HNK5JDMXX:1:2251:27814:30154	83	1	3004896	42	74M	=	3004900	-70	*	*
A00739:196:HNK5JDMXX:2:1166:2184:20087	163	1	3005301	42	66M	=	3005297	61	*	*
A00739:196:HNK5JDMXX:2:1166:2184:20087	83	1	3005297	42	65M	=	3005301	-61	*	*
A00739:196:HNK5JDMXX:1:1264:14136:24627	163	1	3005311	42	57M	=	3005307	52	*	*
A00739:196:HNK5JDMXX:1:1264:14136:24627	83	1	3005307	42	56M	=	3005311	-52	*	*
A00739:196:HNK5JDMXX:1:1277:8992:9737	99	1	3005317	42	51M	=	3005313	46	*	*
A00739:196:HNK5JDMXX:1:1277:8992:9737	147	1	3005313	42	50M	=	3005317	-46	*	*
A00739:196:HNK5JDMXX:1:1206:15013:27085	99	1	3005331	42	97M	=	3005434	199	*	*
A00739:196:HNK5JDMXX:1:1313:12653:10347	163	1	3005400	42	67M	=	3005396	62	*	*
A00739:196:HNK5JDMXX:1:1313:12653:10347	83	1	3005396	42	66M	=	3005400	-62	*	*
A00739:196:HNK5JDMXX:1:1206:15013:27085	147	1	3005434	42	96M	=	3005331	-199	*	*
A00739:196:HNK5JDMXX:1:1309:2609:26459	99	1	3005538	42	97M	=	3005553	111	*	*
A00739:196:HNK5JDMXX:1:1309:2609:26459	147	1	3005553	42	96M	=	3005538	-111	*	*
A00739:196:HNK5JDMXX:1:2266:26883:3427	99	1	3005564	42	97M	=	3005630	162	*	*
A00739:196:HNK5JDMXX:1:2266:26883:3427	147	1	3005630	42	96M	=	3005564	-162	*	*
A00739:196:HNK5JDMXX:1:2319:26910:2440	99	1	3005644	42	97M	=	3005746	198	*	*
A00739:196:HNK5JDMXX:1:2319:26910:2440	147	1	3005746	42	96M	=	3005644	-198	*	*
A00739:196:HNK5JDMXX:2:1185:23194:35618	163	1	3005810	42	89M	=	3005806	84	*	*
A00739:196:HNK5JDMXX:2:1185:23194:35618	83	1	3005806	42	88M	=	3005810	-84	*	*
A00739:196:HNK5JDMXX:1:1403:32380:5243	99	1	3005921	42	97M	=	3005993	168	*	*
A00739:196:HNK5JDMXX:2:2188:12662:17597	99	1	3005921	42	97M	=	3006107	282	*	*
A00739:196:HNK5JDMXX:1:1403:32380:5243	147	1	3005993	42	96M	=	3005921	-168	*	*
A00739:196:HNK5JDMXX:1:2374:21395:32659	99	1	3006104	42	69M	=	3006100	64	*	*
A00739:196:HNK5JDMXX:1:2374:21395:32659	147	1	3006100	42	68M	=	3006104	-64	*	*
A00739:196:HNK5JDMXX:2:2188:12662:17597	147	1	3006107	42	96M	=	3005921	-282	*	*
A00739:196:HNK5JDMXX:2:1271:18674:20901	99	1	3006222	42	97M	=	3006536	410	*	*
A00739:196:HNK5JDMXX:2:1246:2130:31328	99	1	3006486	42	59M	=	3006482	54	*	*
A00739:196:HNK5JDMXX:2:1246:2130:31328	147	1	3006482	42	58M	=	3006486	-54	*	*
A00739:196:HNK5JDMXX:1:1409:22824:33191	163	1	3006524	42	30M	=	3006520	25	*	*
A00739:196:HNK5JDMXX:1:1409:22824:33191	83	1	3006520	42	29M	=	3006524	-25	*	*
A00739:196:HNK5JDMXX:2:1271:18674:20901	147	1	3006536	42	96M	=	3006222	-410	*	*
A00739:196:HNK5JDMXX:2:2174:23647:29199	163	1	3006632	42	39M	=	3006628	34	*	*
A00739:196:HNK5JDMXX:2:2174:23647:29199	83	1	3006628	42	38M	=	3006632	-34	*	*
A00739:196:HNK5JDMXX:2:2201:30246:29481	99	1	3006666	42	97M	=	3006802	232	*	*
A00739:196:HNK5JDMXX:1:1404:29758:16297	163	1	3006707	42	97M	=	3006911	300	*	*
A00739:196:HNK5JDMXX:2:2201:30246:29481	147	1	3006802	42	96M	=	3006666	-232	*	*
A00739:196:HNK5JDMXX:1:1404:29758:16297	83	1	3006911	42	96M	=	3006707	-300	*	*
```

**Comment by taoliu** [comment]
@MikeWLloyd I see. You were using MACS2 v2.2.6. I tried the v2.2.6 and saw the same message as you. I fixed a bunch of errors related to precisions of float numbers in the newest version v2.2.7.1. Could you try to upgrade? The new release can be found on PyPI and bioconda. Will address the zeroDivision warning message in the next version.

**Comment by MikeWLloyd** [comment]
@taoliu I missed that there was an update to the tool! Thanks for being so prompt on helping me resolve this. You are right the updated version fixes the issue.

**Comment by taoliu** [comment]
@MikeWLloyd You are welcome!

**Comment by taoliu** [comment]
ZeroDivision error has been solved in PR #386
